cli/src/lockfiles/parsers/pypi.rs

```diff
@@ -94,3 +114,29 @@ fn package(input: &str) -> Option<PackageDescriptor> {
         package_type: PackageType::PyPi,
     })
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn package_with_extras() {
```

Comment:
An additional [failing] test could be added here to point out an issue:

```rust
        assert_eq!(
            package("git-for-pip-example[PDF] @ git+https://github.com/matiascodesal/git-for-pip-example.git@v1.0.0"),
            Some(PackageDescriptor {
                name: "git-for-pip-example".into(),
                version: "git+https://github.com/matiascodesal/git-for-pip-example.git@v1.0.0".into(),
                package_type: PackageType::PyPi,
            })
        );
```

This test case comes from attempting to replicate an example from [pip install documentation for extras](https://pip.pypa.io/en/stable/cli/pip_install/#examples), [PEP 508 for extras](https://peps.python.org/pep-0508/#extras), and the current (working/valid?) example in `cli/tests/fixtures/complex-requirements.txt`:

```
...
git-for-pip-example @ git+https://github.com/matiascodesal/git-for-pip-example.git@v1.0.0 
[EOF]
```

The test case fails, with the following message:

```sh
running 1 test
thread 'lockfiles::parsers::pypi::test::package_with_extras' panicked at 'assertion failed: `(left == right)`
  left: `Some(PackageDescriptor { name: "git-for-pip-example[pdf]", version: "git+https://github.com/matiascodesal/git-for-pip-example.git@v1.0.0", package_type: PyPi })`,
 right: `Some(PackageDescriptor { name: "git-for-pip-example", version: "git+https://github.com/matiascodesal/git-for-pip-example.git@v1.0.0", package_type: PyPi })`', cli/src/lockfiles/parsers/pypi.rs:142:9
...
```

Reply:
I've added this test case and modified the code to fix it, but this is definitely something we should revisit later. I don't believe that we truly support URL specified dependencies. And if/when we do add support for them, we should revisit how we represent them in the `PackageDescriptor` object. The URL in the `version` field feels unnatural